postgres/tasks_remote.py

```diff
@@ -53,34 +58,87 @@ def setup(ctx):
-    "recreate": "Whether to completely destroy the database prior to loading the data",
+    "add_superuser": "Whether to add a superuser named supersurf that uses the application database password",
     "migrate": "Whether to apply some changes to the snapshot file to migrate from a pre-AWS format"
 })
-def restore_snapshot(ctx, snapshot_name=None, recreate=True, migrate=True):
+def restore_snapshot(conn, snapshot_name=None, add_superuser=True, migrate=True):
     """
     Loads a particular snapshot into the database on AWS through a bastion host
     """
+    if conn.host != conn.config.aws.bastion:
+        raise Exit(f"Did not expect the host {conn.host} while the bastion is {conn.config.aws.bastion}")
 
     snapshot_file_path = download_snapshot(snapshot_name)
 
     # Make minor adjustments to dumps for legacy dumps to work on AWS
     if migrate:
         print("Migrating dump file")
-        ctx.local(f"sed -i 's/OWNER TO surf/OWNER TO postgres/g' {snapshot_file_path}", echo=True)
+        conn.local(f"sed -i 's/OWNER TO surf/OWNER TO postgres/g' {snapshot_file_path}", echo=True)
 
-    # Recreating database with all correct privileges
-    if recreate:
-        print("Recreating databases")
-        setup(ctx)
+    if "test" in conn.config.surfconext.oidc_endpoint:
+        print("Rewriting team ARN's for test communities")
+        production_test_team_urn = "urn:collab:group:surfteams.nl:nl:surfnet:diensten:test_team_zoekportal"
+        development_test_team_urn = "urn:collab:group:test.surfconext.nl:nl:surfnet:diensten:zoekportaal_test_community"
+        conn.local(f"sed -i 's/{production_test_team_urn}/{development_test_team_urn}/g' {snapshot_file_path}",
+                   echo=True)
 
     print("Restoring snapshot")
     # Setup auto-responder
-    postgres_user = ctx.config.django.postgres_user
-    postgres_password = ctx.config.secrets.postgres.password
+    postgres_user = conn.config.django.postgres_user
+    postgres_password = conn.config.secrets.postgres.password
     postgres_password_responder = Responder(pattern=r"Password: ", response=postgres_password + "\n")
     # Run Postgres command with port forwarding
-    with ctx.forward_local(local_port=1111, remote_host=ctx.config.django.postgres_host, remote_port=5432):
-        ctx.local(f"psql -h localhost -p 1111 -U {postgres_user} -W -d edushare -f {snapshot_file_path}",
-                  echo=True, watchers=[postgres_password_responder], pty=True)
+    with conn.forward_local(local_port=1111, remote_host=conn.config.django.postgres_host, remote_port=5432):
+        conn.local(f"psql -h localhost -p 1111 -U {postgres_user} -W -d edushare -f {snapshot_file_path}",
+                   echo=True, watchers=[postgres_password_responder], pty=True)
+
+    if add_superuser:
+        print("Adding superuser")
+        password = conn.config.secrets.postgres_application.password
+        insert_user = insert_django_user_statement("supersurf", password, is_edushare=True)
```

Comment:
Waarom moet er een aparte superuser komen met hetzelfde wachtwoord? Dat betekent dat als je het application wachtwoord weet je ook als superuser zou kunnen authenticeren.
Kunnen we niet de postgres user gebruiken? Of inderdaad supersurf, maar dan met een eigen wachtwoord?

En waarom zit dit in het restore_snapshot command? Moeten we dat niet op een ander moment aanmaken?

Reply:
Als je het applicatie wachtwoord weet, dan kun je sowieso alles wat ook in de admin kan. Admin is een UI laagje op de applicatie. Postgres user kan veel meer (drop, create, extensions). Dat willen we graag afgeschermd houden.

Setup is inderdaad wel logischer. Beetje een brainfart, want dit command riep eerst setup aan. Dan hoeft de add_superuser flag ook niet. Zal het aanpassen als ik weer terug ben.

Reply:
Een eigen wachtwoord kan, maar ik vond alles nu provisionen via Terraform even teveel gedoe. Kunnen we een ticket voor aanmaken.

Reply:
Ja, alleen voor admin UI moet je nog een account en eduvpn hebben.

Ik vind het zeker een goed idee om niet de postgres user te gebruiken met alle rechten. Maar volgens mij is het dan ook weer niet een security best-practice om hetzelfde wachtwoord te gaan gebruiken tussen de applicatie en superuser 😉 . 
Ik snap ook wel dat het weer gedoe is met terraform, dus ik vind het apart hiervan doen prima.